docs/api/services.md

```diff
@@ -0,0 +1,308 @@
+
+State returns the current state of the service, used to print the service in the startup message.
+
+```go
+// State returns the current state of the service.
+func (d *Service) State(ctx context.Context) (string, error)
+```
+
+### Terminate
+
+Terminate terminates the service after the application shuts down using a post shutdown hook, returning an error if it fails.
+
+```go
+// Terminate terminates the service, returning an error if it fails.
+func (d *Service) Terminate(ctx context.Context) error
+```
+
+## Comprehensive Examples
+
+### Example: Adding a service
+
+This example demonstrates how to add a Redis store as a service to the application, backed by the Testcontainers Redis Go module.
+
+```go
+package main
+
+import (
+    "context"
+    "log"
+
+    "github.com/gofiber/fiber/v3"
+    "github.com/redis/go-redis/v9"
+    tcredis "github.com/testcontainers/testcontainers-go/modules/redis"
+)
+
+type redisStore struct {
+    ctr *tcredis.RedisContainer
+}
+
+// Start initializes and starts the service. It implements the [fiber.Service] interface.
+func (s *redisStore) Start(ctx context.Context) error {
+    // start the service
+    c, err := tcredis.Run(ctx, "redis:latest")
+    if err != nil {
+        return err
+    }
+
+    s.ctr = c
+    return nil
+}
+
+// String returns a string representation of the service.
+// It implements the [fiber.Service] interface.
+func (s *redisStore) String() string {
+    return "redis-store"
+}
+
+// State returns the current state of the service.
+// It implements the [fiber.Service] interface.
+func (s *redisStore) State(ctx context.Context) (string, error) {
+    state, err := s.ctr.State(ctx)
+    if err != nil {
+        return "", fmt.Errorf("container state: %w", err)
+    }
+
+    return state.Status, nil
+}
+
+// Terminate stops and removes the service. It implements the [fiber.Service] interface.
+func (s *redisStore) Terminate(ctx context.Context) error {
+    // stop the service
+    return s.ctr.Terminate(ctx)
+}
+
+func main() {
+    cfg := &fiber.Config{}
+
+    // Initialize service.
+    store := &redisStore{}
+    cfg.Services = append(cfg.Services, store)
+
+    app := fiber.New(*cfg)
+
+    ctx := context.Background()
+
+    // Obtain the connection string from the service.
+    connString, err := store.ctr.ConnectionString(ctx)
+    if err != nil {
+        log.Printf("Could not get connection string: %v", err)
+        return
+    }
+
+    // Parse the connection string to create a Redis client.
+    options, err := redis.ParseURL(connString)
+    if err != nil {
+        log.Printf("failed to parse connection string: %s", err)
+        return
+    }
+
+    // Initialize the Redis client.
+    rdb := redis.NewClient(options)
+
+    // Check the Redis connection.
+    if err := rdb.Ping(ctx).Err(); err != nil {
+        log.Fatalf("Could not connect to Redis: %v", err)
+    }
+
+    app.Listen(":3000")
+}
+
+```
+
+### Example: Adding a service with State Management
```

Comment:
Not needed, but it would be cool if the example used the `gofiber/storage driver + testcontainers`. It would make the example so much shorter since the driver already does most of this

Reply:
On it!